ci-operator/step-registry/cucushift/upgrade/vsphere-hw/cucushift-upgrade-vsphere-hw-commands.sh

```diff
@@ -0,0 +1,128 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+if [[ -f "${SHARED_DIR}/kubeconfig" ]] ; then
+    export KUBECONFIG=${SHARED_DIR}/kubeconfig
+fi
+
+# Setup proxy if it's present in the shared dir
+if [[ -f "${SHARED_DIR}/proxy-conf.sh" ]] 
+then
+    # shellcheck disable=SC1091
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+source "${SHARED_DIR}/govc.sh"
+# list all the virtual machines in the folder/rp
+cluster_name=$(<"${SHARED_DIR}"/clustername.txt)
+vm_path="/${GOVC_DATACENTER}/vm/${cluster_name}"
+clustervms=$(govc ls "${vm_path}-*")
+
+first_vm=$(echo "${clustervms}" | cut -d" " -f1)
+current_hw_version=$(govc vm.info -json=true "${first_vm}" | jq -r .VirtualMachines[0].Config.Version)
+echo "Current hardware version is: ${current_hw_version}"
+hw_version=$(echo "${current_hw_version}" | cut -d'-' -f2)
+
+if (( hw_version >= 15 )); then
+    echo "VM hardware version is not less than 15, no upgrade is required"
+    exit 0
+fi
+
+master_nodes=$(oc get nodes -l node-role.kubernetes.io/master -o jsonpath='{range .items[*]}{.metadata.name}{"\n"}{end}')
+worker_nodes=$(oc get nodes -l node-role.kubernetes.io/worker -o jsonpath='{range .items[*]}{.metadata.name}{"\n"}{end}')
+
+echo "Updating VM hardware version"
+# shellcheck disable=SC2068
+for node in ${master_nodes[@]} ${worker_nodes[@]}; do
+    echo "Marking the node as unschedulable"
+    oc adm cordon "${node}"
+    if [[ "${worker_nodes[*]}" =~ ${node} ]]; then
+        echo "Evacuate the pods from the compute node"
+        oc adm drain "${node}" --force=true --delete-emptydir-data --ignore-daemonsets
+    fi
+    echo "Power off VM"
+    govc vm.power -s=true ${vm_path}/${node}
+
+    echo "Check VM is powered off"
+    timeout=10
+    while (( timeout > 0 )); do
+        sleep 1m
+        (( timeout -= 1))
+        status=$(govc vm.info -json ${vm_path}/${node} | jq -r .VirtualMachines[].Runtime.PowerState)
+
+        if [[ "${status}" == "poweredOff" ]]; then
+            break
+        fi
+    done
+
+    if [[ "${status}" != "poweredOff" ]]; then 
+        echo >&2 "Shutdown VM failed"
+        oc describe node/${node}
+        exit 1
+    fi
+
+    echo "Upgrade VM hardware version to 15"
+    govc vm.upgrade -vm ${vm_path}/${node} -version=15
```

Comment:
how about to introduce a new env var for hardware version instead of hardcode 15 here?

Reply:
Good suggestion! Updated. Thanks!